crates/codegen/parser/src/chumsky/code_generator.rs

```diff
@@ -166,48 +157,36 @@ impl CodeGenerator {
                 &output_dir.join("parse.rs"),
                 &format!(
                     "{}
-                
-                    #[allow(dead_code)]
-                    pub struct Parsers {{
-                        {}
-                    }}
 
-                    impl Parsers {{
-                        pub fn new(version: &Version) -> Self {{
-                            // Declare all versions -----------------------------
+                    pub fn create_parsers(version: &Version) -> BTreeMap<ProductionKind, Parser> {{
+                        let mut parsers: BTreeMap<ProductionKind, Parser> = BTreeMap::new();
+                        
+                        // Declare all versions -----------------------------
 
-                            {}
-
-                            // Declare all productions --------------------------
+                        {}
 
-                            type DeferredLexParserType = Recursive<'static, char, Option<Rc<lex::Node>>, ErrorType>;
-                            type DeferredCSTParserType = Recursive<'static, char, Option<Rc<cst::Node>>, ErrorType>;
+                        // Declare all productions --------------------------
 
-                            {}
+                        {}
 
-                            // Macros -------------------------------------------
+                        // Macros -------------------------------------------
 
-                            {}
+                        {}
 
-                            // Define all productions ---------------------------
+                        // Define all productions ---------------------------
 
-                            {}
+                        {}
```

Comment:
nit: suggestion on using named arguments in long `format!()` expressions, to make it a bit easier to see what is getting added/removed:

```rust
format!("
     // Declare all versions -----------------------------

     {versions}

     // Declare all productions --------------------------

     {productions}
",
    versions = get_versions(),
    productions = get_productions(),
);
```

Reply:
If you agree, a few other places in `codegen_parser` can benefit from this. Not blocking for this PR though.

Reply:
sure, agreed

Reply:
[#226] (https://github.com/NomicFoundation/slang/issues/266)